### ISSUE TITLE
group install

### ISSUE THREAD

**Issue by burlen** [LBL-EESA/TECA#524]
test if I can install to Travis's install

`/global/project/projectdirs/m1517/cascade/software_m1517_build/TECA`

**Comment by burlen** [comment]
Do not share installs on `/project` due to poor Python performance

**Comment by burlen** [comment]
![install_fs_par](https://user-images.githubusercontent.com/115053/105529335-19f36580-5c9b-11eb-8345-b5d23a025546.png)

**Comment by burlen** [comment]
In a follow up I made the runs from scratch and common consecutively on the same set of nodes. 2 sets of runs were made. There was very little difference in performance. 

## Run 1
### common
real	36m47.406s
real	0m3.409s
real	2m14.987s
real	0m18.881s
real	2m16.241s
real	0m18.924s
real	0m48.318s

### scratch
real	33m57.304s
real	0m9.579s
real	2m14.957s
real	0m24.779s
real	2m26.149s
real	0m19.955s
real	1m2.819s

## Run 2
### common
real	33m25.470s
real	0m3.253s
real	3m19.072s
real	0m18.403s
real	2m17.176s
real	0m18.852s
real	0m54.915s
### scratch
real	33m19.508s
real	0m10.048s
real	3m17.921s
real	0m27.116s
real	2m30.075s
real	0m21.127s
real	1m2.247s

**Comment by burlen** [comment]
@taobrienlbl what do you think about the following layout for the installs in `/global/common/software/m1517`?

the root of the teca installs: `/global/common/software/m1517/teca`
release of TECA version X.Y.Z: `/global/common/software/m1517/teca/X.Y.Z`
A symlink to the latest release:  `/global/common/software/m1517/teca/stable`
rolling development installs:  `/global/common/software/m1517/teca/develop-<YYYY_MM_DD>`
a symlink to the latest development install:  `/global/common/software/m1517/teca/develop`

I have been doing haswell based installs which run on both haswell and knl architectures. Do you think its worth it to have KNL specific installs too? I hadn't gone that route because cross compiling dependencies like Python and HDF5 for KNL didn't work.

**Comment by taobrienlbl** [comment]
That layout seems sensible to me.  Would it be worth considering using `develop-<GITHASH>` instead of `develop-<YYYY_MM_DD>` (or using a symlink so that we have both directories present?)?

One thing to consider is that we have very limited space in our quota for `/global/common/software/m1517`: 10GB.  The most recent TECA + superbuild install occupies 1.1GB, so we would have to take care not to have too many installs.  If we factor the superbuild out into a different install directory, I suspect the TECA install itself should be substantially smaller.  The modulefile would have to be a bit more complicated to accommodate this, but I think it is doable.

And I think haswell-only is fine.  IIRC, I did a knl-specific compile at one point and there was no discernible difference in performance.

**Comment by burlen** [comment]
Thanks for the feedback. 

Using the sha instead of the date is a good idea! the superbuild configure script could get the latest sha for use the in the CMAKE_INSTALL_PREFIX by something like
```
sha=`git ls-remote git@github.com:LBL-EESA/TECA.git | grep HEAD |  cut -c1-8`
```

I've documented "m1517 installs" in the rtd user guide (see #541). There I mention only 2 installs `stable` and `develop` where `stable` is a symlink to the latest release and `develop` is a symlink to an unreleased development install.

Given the quota issues, we may have to do what you suggest installing dependencies separately, or limit ourselves to two installs, rm'ing all but the latest `stable` and `develop` install as we roll out the next. Either way, I've only documented the symlinks so our decisions as to how things land on disk should be invisible to users.
